src/mahoji/lib/abstracted_commands/naxxusCommand.ts

```diff
@@ -0,0 +1,240 @@
+	weapon: 'Drygore rapier',
+	shield: 'Offhand drygore rapier',
+	ring: 'Ignis ring(i)'
+});
+
+const itemBoosts: {
+	item: Item;
+	boost: number;
+	setup: 'mage' | 'melee';
+}[] = [
+	{
+		item: getOSItem('Void Staff'),
+		boost: 10,
+		setup: 'mage'
+	},
+	{
+		item: getOSItem('Abyssal Tome'),
+		boost: 7.5,
+		setup: 'mage'
+	},
+	{
+		item: getOSItem('Spellbound Ring(i)'),
+		boost: 2.5,
+		setup: 'mage'
+	},
+	{
+		item: getOSItem('Spellbound Ring'),
+		boost: 2,
+		setup: 'mage'
+	}
+];
+
+const NAXXUS_HP = 3900;
+
+function calcSetupPercent(
+	maxStats: GearStats,
+	userStats: GearStats,
+	heavyPenalizeStat: keyof GearStats,
+	ignoreStats: (keyof GearStats)[]
+) {
+	let numKeys = 0;
+	let totalPercent = 0;
+	for (const [key, val] of Object.entries(maxStats) as [keyof GearStats, number][]) {
+		if (val <= 0 || ignoreStats.includes(key)) continue;
+		const rawPercent = Math.min(100, calcWhatPercent(userStats[key], val));
+		totalPercent += rawPercent;
+		numKeys++;
+	}
+	totalPercent /= numKeys;
+	// Heavy penalize for having less than 50% in the main stat of this setup.
+	if (userStats[heavyPenalizeStat] < maxStats[heavyPenalizeStat] / 2) {
+		totalPercent = Math.floor(Math.max(0, totalPercent / 2));
+	}
+	return totalPercent;
+}
+
+export async function naxxusCommand(user: KlasaUser, channelID: bigint, quantity: number | undefined) {
+	const [hasReqs, rejectReason] = user.hasMonsterRequirements(Naxxus);
+	if (!hasReqs) {
+		return `${user.username} doesn't have the requirements for this monster: ${rejectReason}`;
+	}
+
+	const boosts = [];
+	let effectiveTime = Naxxus.timeToFinish;
+	if (isWeekend()) {
+		effectiveTime = reduceNumByPercent(effectiveTime, 5);
+		boosts.push('5% Weekend boost');
+	}
+
+	if ((user.settings.get(UserSettings.MonsterScores)[Naxxus.id] ?? 0) < 100) {
+		effectiveTime = increaseNumByPercent(effectiveTime, 15);
+		boosts.push('-15% <100 KC');
+	}
+
+	if ((user.settings.get(UserSettings.MonsterScores)[Naxxus.id] ?? 0) > 500) {
+		effectiveTime = reduceNumByPercent(effectiveTime, 15);
+		boosts.push('15% >500 KC');
+	}
+
+	const mage = calcSetupPercent(bisMageGear.stats, user.getGear('mage').stats, 'attack_magic', [
+		'attack_stab',
+		'attack_slash',
+		'attack_crush',
+		'attack_ranged',
+		'defence_crush',
+		'defence_ranged',
+		'defence_slash',
+		'defence_stab'
+	]);
+
+	const melee = calcSetupPercent(bisMeleeGear.stats, user.getGear('melee').stats, 'attack_stab', [
+		'attack_slash',
+		'attack_crush',
+		'attack_ranged',
+		'attack_magic',
+		'defence_magic'
+	]);
+
+	const totalGearPercent = (melee + mage) / 2;
+	const speedReductionMaxPercent = 15;
+	effectiveTime = reduceNumByPercent(
+		effectiveTime,
+		calcPercentOfNum(Math.ceil(totalGearPercent), speedReductionMaxPercent)
+	);
+	boosts.push(
+		`${calcPercentOfNum(Math.ceil(totalGearPercent), speedReductionMaxPercent)}% gear stats [${Math.ceil(
+			melee
+		)}% of max melee, ${Math.ceil(mage)}% of max mage]`
+	);
+
+	itemBoosts.forEach(itemBoost => {
```

Comment:
we dont use `forEach` anywhere, use `for of`

Reply:
Swapped to `for of`